tool/state-processor/src/system/mod.rs

```diff
@@ -1,5 +1,7 @@
 // darwinia
 use crate::*;
+use sp_core::U256;
```

Comment:
Can be replaced with `[u64; 4]`

Reply:
`Approval { amount: v.as_u128(), deposit: 0 },`

There is a u128 conversion afterward, Use U256 would be much easier to achieve this. Or we need to do bit operation, easy to make mistake.

Reply:
Try `primitive-types    = { version = "0.12" }`.

I don't want to introduce too many unrelated items.

Reply:
Let me have a try.

Reply:
```
error[E0277]: the trait bound `U256: WrapperTypeDecode` is not satisfied
   --> src/system/mod.rs:153:7
    |
150 |                     self.solo_state.take_map(
    |                                     -------- required by a bound introduced by this call
...
153 |                         &mut approves,
    |                         ^^^^^^^^^^^^^ the trait `WrapperTypeDecode` is not implemented for `U256`
    |
    = help: the following other types implement trait `WrapperTypeDecode`:
              Arc<T>
              Box<T>
              Rc<T>
    = note: required for `U256` to implement `Decode`
note: required by a bound in `State::take_map`
   --> src/main.rs:199:6
    |
191 |     fn take_map<D, F>(
    |        -------- required by a bound in this
...
199 |         D: Decode,
    |            ^^^^^^ required by this bound in `State::take_map`
```

Unlucky. The `U256` in the `primitive-types` does implement Codec traits.

Reply:
`features = ["codec"]`, `sp_core` is just r-exporting this.

Reply:
It works. Thanks !!!!!